networkx/generators/classic.py

```diff
@@ -322,7 +322,7 @@ def grid_2d_graph(m,n,periodic=False,create_using=None):
         boundary nodes via periodic boundary conditions.
     """
     G=empty_graph(0,create_using)
-    G.name="grid_2d_graph"
+    G.name="grid_2d_graph(%d,%d)"%(m,n)
```

Comment:
Yes, we've started supporting both Python 2 and 3 since awhile. You should also consider revising new lines 387, 403, 449, 466, 495.